### PR TITLE
Fix parsing of legacy DomRange models

### LINKED ISSUE #141: Fix backward-compatibility of Locator's DOMRange
1.x used to parse a DOM range char offset as `offset`, but in 2.0 we're using `charOffset` ([per spec](https://github.com/readium/architecture/blob/master/models/locators/extensions/html.md#the-start-and-end-object)). We need to fix the parsing to support both cases to make sure we don't break existing implementations having persisted `Locator`s with `offset`.

### PR DESCRIPTION
**Pull request by mickael-menu**
Fix #141 

@aferditamuriqi I added some unit test but I still would like to add a test with a full legacy `Locator` to make sure that we parse them correctly. Could you leave a sample in a comment? Thanks

### CODE REVIEW

File: r2-shared/src/main/java/org/readium/r2/shared/publication/html/DomRange.kt [hunk 1]
```diff
@@ -88,6 +88,10 @@ data class DomRange(
                     cssSelector = cssSelector,
                     textNodeIndex = textNodeIndex,
                     charOffset = json.optPositiveInt("charOffset")
+                        // The model was using `offset` before, so we still parse it to ensure
+                        // backward-compatibility for reading apps having persisted legacy Locator
+                        // models.
+                        ?: json.optPositiveInt("offset")
```
Comment (aferditamuriqi) [inline r2-shared/src/main/java/org/readium/r2/shared/publication/html/DomRange.kt:94]: @mickael-menu i am still trying to find out why this is not working, i basically am getting no offset at all. i'll update if i find something.

Comment (aferditamuriqi) [reply r2-shared/src/main/java/org/readium/r2/shared/publication/html/DomRange.kt:94]: ok, this seems to work:

```
            fun fromJSON(json: JSONObject?, warnings: WarningLogger? = null): Point? {
                
                val cssSelector = json?.optNullableString("cssSelector")
                
                val textNodeIndex = json?.optInt("textNodeIndex")

                val charOffset = json?.optInt("charOffset") ?: json?.optInt("offset")

                if (cssSelector == null || textNodeIndex == null) {
                    warnings?.log(Point::class.java, "[cssSelector] and [textNodeIndex] are required", json)
                    return null
                }
                
                return Point(
                    cssSelector = cssSelector,
                    textNodeIndex = textNodeIndex,
                    charOffset = charOffset
                )
            }

```

Comment (aferditamuriqi) [reply r2-shared/src/main/java/org/readium/r2/shared/publication/html/DomRange.kt:94]: but it's not fully solved with this, i need to figure out a bit more.

**Review by aferditamuriqi — APPROVED** [review]
this seems to work, i needed to still add the offset in the toJSON method though, otherwise previous versions would crash 
            put("offset", charOffset)

**Comment by mickael-menu** [comment]
@aferditamuriqi I'm a bit wary of adding `offset` in `toJSON` because that would mean `charOffset` and `offset` becoming aliases in the model.

I tried making highlights at different locations in an EPUB resource from `master`, and they still worked after upgrading to `develop`+`fix/legacy-dom-ranges`. Maybe you have a private part in your codebase referencing `offset`?

**Pull request merged**